javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -247,6 +247,32 @@ public static boolean isSelfAccess(ExpressionTree tree) {
     return false;
   }
 
+  /**
+   * Returns true iff the underlying JCTree exists and start at a position p &le; 0 in the source
```

Comment:
This method could be PR'd right away separately. It probably should be reviewed separately from this PR, anyway, since it will become part of the javacutil API. When you create the PR for this method, you could point at the uses of this method in this PR as justification for adding a new API method.